zerver/actions/invites.py

```diff
@@ -310,22 +314,16 @@ def do_invite_users(
         event = {
             "prereg_id": prereg_user.id,
             "referrer_id": user_profile.id,
-            "email_language": user_profile.realm.default_language,
+            "email_language": realm.default_language,
             "invite_expires_in_minutes": invite_expires_in_minutes,
         }
-        queue_json_publish("invites", event)
+        queue_event_on_commit("invites", event)
 
-    if skipped:
-        raise InvitationError(
-            _(
-                "Some of those addresses are already using Zulip, "
-                "so we didn't send them an invitation. We did send "
-                "invitations to everyone else!"
-            ),
-            skipped,
-            sent_invitations=True,
-        )
-    notify_invites_changed(user_profile.realm, changed_invite_referrer=user_profile)
+    transaction.on_commit(
+        lambda: notify_invites_changed(realm, changed_invite_referrer=user_profile)
```

Comment:
Usually our pattern is to make a function like `notify_invites_changed` do the `transaction.on_commit` part internally.

Reply:
It's not always called from within a transaction, however.

Reply:
Yes but I believe `transaction.on_commit` just gets called immediately when outside one; and probably we will eventually transaction-ify all of our `send_event` callers.